edge/pkg/devicetwin/dttype/types_test.go

```diff
@@ -0,0 +1,659 @@
+	if err != nil {
+		panic(err)
+	}
+	return keyErrorTwinUpdate, bytesTwinKeyError
+}
+
+// createTwinUpdate() is function to create DeviceTwinUpdate with correct actual and expected values.
+func createTwinUpdate() (DeviceTwinUpdate, []byte) {
+	var keyTwinUpdate DeviceTwinUpdate
+	twinKey := make(map[string]*MsgTwin)
+	var expected TwinValue
+	var actual TwinValue
+	value := "value"
+	valueMetaData := &ValueMetadata{Timestamp: time.Now().UnixNano() / 1e6}
+	expected.Value = &value
+	expected.Metadata = valueMetaData
+	actual.Value = &value
+	actual.Metadata = valueMetaData
+	twinKey["key1"] = &MsgTwin{Expected: &expected, Actual: &actual}
+	keyTwinUpdate.Twin = twinKey
+	bytesTwinKey, err := json.Marshal(keyTwinUpdate)
+	if err != nil {
+		panic(err)
+	}
+	return keyTwinUpdate, bytesTwinKey
+}
+
+// createTwinUpdateWrongActual() is function to create  DeviceTwinUpdate having right key with wrong actual value.
+func createTwinUpdateWrongActual() (DeviceTwinUpdate, []byte) {
+	var keyTwinUpdateActualValueError DeviceTwinUpdate
+	twinKeyActualValueError := make(map[string]*MsgTwin)
+	var actualValueErrorExpected TwinValue
+	var actualValueErrorActual TwinValue
+	valueExpected := "value"
+	valueActual := "value~"
+	valueMetaDataActualValueError := &ValueMetadata{Timestamp: time.Now().UnixNano() / 1e6}
+	actualValueErrorExpected.Value = &valueExpected
+	actualValueErrorExpected.Metadata = valueMetaDataActualValueError
+	actualValueErrorActual.Value = &valueActual
+	actualValueErrorActual.Metadata = valueMetaDataActualValueError
+	twinKeyActualValueError["key1"] = &MsgTwin{Expected: &actualValueErrorExpected, Actual: &actualValueErrorActual}
+	keyTwinUpdateActualValueError.Twin = twinKeyActualValueError
+	bytesTwinKeyActualValueError, err := json.Marshal(keyTwinUpdateActualValueError)
+	if err != nil {
+		panic(err)
+	}
+	return keyTwinUpdateActualValueError, bytesTwinKeyActualValueError
+}
+
+// createTwinUpdateWrongExpected() is function to create DeviceTwinUpdate having right key with wrong expected value.
+func createTwinUpdateWrongExpected() (DeviceTwinUpdate, []byte) {
+	var keyTwinUpdateExpectedValueError DeviceTwinUpdate
+	twinKeyExpectedValueError := make(map[string]*MsgTwin)
+	var expectedValueErrorExpected TwinValue
+	var expectedValueErrorActual TwinValue
+	valueExpectedValueError := "value~"
+	valueMetaDataExpectedValueError := &ValueMetadata{Timestamp: time.Now().UnixNano() / 1e6}
+	expectedValueErrorExpected.Value = &valueExpectedValueError
+	expectedValueErrorExpected.Metadata = valueMetaDataExpectedValueError
+	expectedValueErrorActual.Value = &valueExpectedValueError
+	expectedValueErrorActual.Metadata = valueMetaDataExpectedValueError
+	twinKeyExpectedValueError["key1"] = &MsgTwin{Expected: &expectedValueErrorExpected, Actual: &expectedValueErrorActual}
+	keyTwinUpdateExpectedValueError.Twin = twinKeyExpectedValueError
+	bytesTwinKeyExpectedValueError, err := json.Marshal(keyTwinUpdateExpectedValueError)
```

Comment:
since we are building this object ourselves , we don't expect a marshal error here. panic is not needed.